aviatrix/resource_aviatrix_dcf_tls_profile_test.go

```
@@ -0,0 +1,165 @@
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckDcfTLSProfileBasic() string {
+	return `
+resource "aviatrix_dcf_tls_profile" "test" {
+	display_name           = "test-dcf-tls-profile"
+	certificate_validation = "CERTIFICATE_VALIDATION_LOG_ONLY"
+	verify_sni            = true
+}
+`
+}
+
+func testAccCheckDcfTLSProfileUpdate() string {
+	return `
+resource "aviatrix_dcf_tls_profile" "test" {
+	display_name           = "test-dcf-tls-profile-updated"
+	certificate_validation = "CERTIFICATE_VALIDATION_ENFORCE"
+	verify_sni            = false
+}
+`
+}
+
+func testAccCheckDcfTLSProfileWithCABundle() string {
+	return `
+resource "aviatrix_dcf_tls_profile" "test" {
+	display_name           = "test-dcf-tls-profile-with-ca"
+	certificate_validation = "CERTIFICATE_VALIDATION_ENFORCE"
+	verify_sni            = true
+	ca_bundle_id          = "test-ca-bundle-uuid"
+}
+`
+}
+
+func testAccCheckDcfTLSProfileExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("no DCF TLS Profile resource found: %s", n)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no DCF TLS Profile ID is set")
+		}
+
+		client, ok := testAccProviderVersionValidation.Meta().(*goaviatrix.Client)
+		if !ok {
+			return fmt.Errorf("failed to assert Meta as *goaviatrix.Client")
+		}
+
+		_, err := client.GetTLSProfile(context.Background(), rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("failed to get DCF TLS Profile status: %w", err)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckDcfTLSProfileDestroy(s *terraform.State) error {
+	client, ok := testAccProviderVersionValidation.Meta().(*goaviatrix.Client)
+	if !ok {
+		return fmt.Errorf("failed to assert Meta as *goaviatrix.Client")
+	}
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aviatrix_dcf_tls_profile" {
+			continue
+		}
+
+		_, err := client.GetTLSProfile(context.Background(), rs.Primary.ID)
```

I don't think this is a great pattern. Can you try this instead?

1) Create a helper function, `generateAcceptanceCheckFunctionDcfTLSProfileDestroy`, that takes in a `testing.T` and returns a `func(*terraform.State) error`. 
2) In that helper function, create a closure that runs `t.Context()` to get a test-scoped context for use in these calls, and then return the contents of this helper function with this `context.Background` swapped out.

I'm ambivalent about returning an error on casting failure -- if this absolutely should never happen, you might also want to use `testify.Assert` with the `testing.T` instead of returning the `failed to assert Meta as *goaviatrix.Client` error.

Also update the other helper functions here to use a similar pattern.

---

Also, is the return value here really irrelevant so long as it isn't a failure, or should you also be doing sanity checks to make sure this matches your expectations?

---

looking at other tests this is how it seems to be done

---

Right, I just don't like using `context.Background()` if there's a reasonable way to provide a context that will limit the scope down to the lifespan of the test. We already use the closure pattern elsewhere, this would just use it to give a testing.T-scoped context for this call.

---

created a context.Withtimeout for one, and t.Context() for the other